src/backend/distributed/commands/utility_hook.c

```diff
@@ -255,9 +256,15 @@ citus_ProcessUtility(PlannedStmt *pstmt,
 
 	if (!CitusHasBeenLoaded())
 	{
-		if (!IsMainDB)
+		if (!IsMainDB &&
```

Comment:
Could use a comment what we're doing her. Something as simle as:
> We always execute CREATE DATABASE from the main database. Since it's non-transactional there are no transactional visibility issues that cause issues when doing that. And this way we only have to consider one codepath when creating databases.